### PR TITLE
Map several registry-related classes.

### PR DESCRIPTION
**Pull request by LambdAurora**

### CODE REVIEW

File: mappings/net/minecraft/util/registry/SimpleRegistry.mapping [hunk 1]
```diff
@@ -7,7 +7,7 @@ CLASS net/minecraft/unmapped/C_secmvxxe net/minecraft/util/registry/SimpleRegist
 	FIELD f_oyicsmaj byId Ljava/util/Map;
 	FIELD f_rbivtjbc nextId I
 	FIELD f_tjecakfq lifecycle Lcom/mojang/serialization/Lifecycle;
-	FIELD f_tlngcayg entryToLifecycle Ljava/util/Map;
+	FIELD f_tlngcayg getLifecycle Ljava/util/Map;
```
Comment (LambdAurora) [inline mappings/net/minecraft/util/registry/SimpleRegistry.mapping]:
```suggestion
	FIELD f_tlngcayg entryToLifecycle Ljava/util/Map;
```

Damn you enigma.

File: mappings/net/minecraft/data/server/tag/BlockTagsProvider.mapping [hunk 2]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/unmapped/C_wayypnpx net/minecraft/data/server/tag/BlockTagsProvider
```
Comment (EnnuiL) [inline mappings/net/minecraft/data/server/tag/BlockTagsProvider.mapping:1]: Oh, I've touched those providers on #267; I feel like using `BaseBlocksTagProvider` and `FutureBlockTagsProvider` might be better than `BlocksTagProvider` and `UpdateOneTwentyBlockTagsProvider`; i really don't like the mojmap name for the latter (although to be fair, it's only an opinion that might be bad if they add concurrent ones) and you'll might want consistency with the item providers, which had the same split except that they both share a common source that isn't abstract either

**Review by EnnuiL — CHANGES_REQUESTED** [review]
The PR seems good, even though I'm not really a mappings person; I did notice, however, some spots in which I'm better equipped to talk about

**Review by OroArmor — APPROVED** [review]

**Review by NoComment1105 — APPROVED** [review]

File: mappings/net/minecraft/data/server/tag/BlockTagsProvider.mapping (hunk 2 above)
Comment (LambdAurora) [reply mappings/net/minecraft/data/server/tag/BlockTagsProvider.mapping:1]: I'm not entirely sure about the FutureBlockTagsProvider

Even if then in 1.20 there's one for 1.21 stuff, the mapping will be lost due to mojmap having very different names.
I'm also wondering what would happens if there's ever 2 concurrent experiments (which kinda already happened with the combat updates, etc)?

Comment (EnnuiL) [reply mappings/net/minecraft/data/server/tag/BlockTagsProvider.mapping:1]: yeah, after thinking about it, `FutureBlockTagsProvider` wouldn't really be great; The rest of the feedback might still be worth investigating though, but yeah, maybe `UpdateOneTwenty` is the better name for now (unless Mojang ends revealing that update name :P)

**Pull request merged**